examples/paho-mqtt/main.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (C) 2019 Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file        main.c
+ * @brief       Example using MQTT Paho package from RIOT
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include "xtimer.h"
+#include "shell.h"
+#include "thread.h"
+#include "mutex.h"
+#include "mqtt.h"
+#include "MQTTClient.h"
+
+#define BUF_SIZE                        1024
+#define MQTT_VERSION_v311               4       /* MQTT v3.1.1 version is 4 */
+#define COMMAND_TIMEOUT_MS              4000
+
+#ifndef DEFAULT_MQTT_CLIENTID
+#define DEFAULT_MQTT_CLIENTID           "riot"
+#endif
+
+#ifndef DEFAULT_MQTT_USER
+#define DEFAULT_MQTT_USER               "riot"
```

Comment:
If default mqtt user is empty, does this switch to anonymous authentication ? If yes, I would use an empty string as default username.

Reply:
I'm sorry, this question was not answered. Let me rephrase it.
Do you think that when the default mqtt user is set to an empty string the anonymous authentication is used ? If it's the case, I would use:
```
#define DEFAULT_MQTT_USER               ""
```
instead. This way, by default, the anonymous authentication will be used if no username is given.

Reply:
Important questions. What I have seen in the test is that if the `mosquitto` broker has a password file for user authentication, the anonymous authentication only works if there is no user name.

Reply:
sorry @aabadie , I must have missed that. Interesting point @gschorcht and @aabadie . I'm gonna leave it empty and modify/clarify a bit on the README